src/applications/disability-benefits/all-claims/utils/validations.js

```diff
@@ -0,0 +1,241 @@
+    missingvalue: 'Service date missing month, day, or year',
+  };
+
+  // Range validations
+  if (!isValidDateRange(startDateApproximate, endDateApproximate)) {
+    errors.endDate.addError(messages.range);
+  }
+
+  if (
+    endDateApproximate.year.value < 1990 ||
+    (endDateApproximate.year.value === 1990 &&
+      endDateApproximate.day.value < 8 &&
+      endDateApproximate.month.value < 2)
+  ) {
+    errors.endDate.addError(messages.enddate);
+  }
+
+  if (startDateApproximate > moment()) {
+    errors.startDate.addError(messages.futuredate);
+  }
+
+  if (endDateApproximate > moment()) {
+    errors.endDate.addError(messages.futuredate);
+  }
+
+  // Validations against empty date object due to missing value
+  if (
+    !startDateApproximate.day.value ||
+    !startDateApproximate.month.value ||
+    !startDateApproximate.year.value
+  ) {
+    errors.startDate.addError(messages.missingvalue);
+  }
+
+  if (
+    !endDateApproximate.day.value ||
+    !endDateApproximate.month.value ||
+    !endDateApproximate.year.value
+  ) {
+    errors.endDate.addError(messages.missingvalue);
+  }
+}
+
+export function validateToxicExposureGulfWar2001Dates(
+  errors,
+  { startDate, endDate },
+) {
+  const startDateApproximate = convertToDateField(startDate);
+  const endDateApproximate = convertToDateField(endDate);
+
+  const messages = {
+    range: 'Your service start date must be before your end date',
+    enddate: 'Service end date must be after September 11, 2001',
+    // This is largely redundant because the component validation already prevents
+    // inputting dates in the future. However, this is in place just in case, because...
+    // TODO: The current schema validation says that the Veteran can input a year
+    // up to 100 years in the future. That is incorrect and should be corrected/removed.
+    futuredate: 'Service date can not be in the future',
+    // Schema validations will display an error when a value is missing, however it
+    // will not properly prevent the Veteran from progressing, necessitating this.
+    // TODO: The error message here can be made more specific, but that will somewhat
+    // depend on what decisions (and schema updates) are made in regards to partial dates.
+    missingvalue: 'Service date missing month, day, or year',
+  };
+
+  // Range validations
+  if (!isValidDateRange(startDateApproximate, endDateApproximate)) {
+    errors.endDate.addError(messages.range);
+  }
+
+  if (
+    endDateApproximate.year.value < 2001 ||
+    (endDateApproximate.year.value === 2001 &&
+      endDateApproximate.day.value < 11 &&
+      endDateApproximate.month.value < 9)
+  ) {
+    errors.endDate.addError(messages.enddate);
+  }
+
+  if (startDateApproximate < moment()) {
+    errors.startDate.addError(messages.futuredate);
+  }
+
+  if (endDateApproximate < moment()) {
+    errors.endDate.addError(messages.futuredate);
+  }
```

Comment:
Must the service end date be in the past?

Reply:
Wasn't sure if 'service date' could apply to only a single service date range, if a Veteran had multiple service dates.